full-node/sov-ethereum/src/lib.rs

```diff
@@ -46,21 +47,32 @@ pub mod experimental {
     }
 
     impl Ethereum {
-        fn make_raw_tx(&self, evm_tx: EvmTransaction) -> Result<Vec<u8>, std::io::Error> {
+        fn make_raw_tx(
+            &self,
+            raw_tx: RawEvmTransaction,
+        ) -> Result<(H256, Vec<u8>), jsonrpsee::core::Error> {
+            let signed_transaction: RethTransactionSignedNoHash =
+                raw_tx.clone().try_into().map_err(EthApiError::from)?;
+
+            let tx_hash = signed_transaction.hash();
+            let sender = signed_transaction
+                .recover_signer()
+                .ok_or(EthApiError::InvalidTransactionSignature)?;
+
             let mut nonces = self.nonces.lock().unwrap();
             let nonce = *nonces
-                .entry(evm_tx.sender)
+                .entry(sender.into())
```

Comment:
Are those EVM nonces or rollup nonces? (Or are they the same?!)

1. Maybe better to have some concurrent lock-less data structure that can operate on separate keys concurrently. Some options [chashmap](https://docs.rs/chashmap/latest/chashmap/), [dashmap](https://github.com/xacrimon/dashmap), [leapfrog](https://github.com/robclu/leapfrog) ?
2. If it is EVM-based nonce - Then we shouldn't even modify EVM based transaction? (Or should it be updated after EVM processes a block?)
3. If it is rollup nonce, it also should probably be handled differently on rollup level and not in EVM module?
3.

Reply:
1. `dashmap` can deadlock occasionally, `chashmap` has only 4 stars on gitlab and wasn't updated since long time.  [leapfrog](https://github.com/robclu/leapfrog) looks ok but I will need spent more time to examine it further, so will be done in another PR. 
2. rollup nonce.
3.  This code is outside if the EVM module (in `full- node/sov-ethereum`). It is our temporary way for wrapping evm transactions inside rollup transactions (will be handled by sequencer when we have it)

Reply:
We can look into other packages, it was just one-minute google to see what can be available. Generally in .net projects we tend to use built in: https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-7.0 ,  I expected something similar on rust side.